eqcorrscan/core/match_filter/matched_filter.py

```diff
@@ -701,8 +709,16 @@ def match_filter(template_names, template_list, st, threshold,
     if str(threshold_type) == str("absolute"):
         thresholds = [threshold for _ in range(len(cccsums))]
     elif str(threshold_type) == str('MAD'):
-        thresholds = [threshold * np.median(np.abs(cccsum))
-                      for cccsum in cccsums]
+        if cores:
+            median_cores = min([cores, len(cccsums)])
+            if len(cccsums) * len(cccsums[0]) < 2e7:  # parallel not worth it
+                median_cores = 1
+            medians = Parallel(n_jobs=median_cores)(delayed(
+                _mad)(cccsum) for cccsum in cccsums)
```

Comment:
```suggestion
            with ThreadPoolExecutor(max_workers=median_cores) as executor:  
               # Because numpy releases GIL threading can use multiple cores
               medians = executor.map(_mad, cccsums)
```